File: SkillMatchBackEnd/src/main/java/pattern/skillmatchbackend/persistenza/dao/Annuncio.java
package pattern.skillmatchbackend.persistenza.dao;
import pattern.skillmatchbackend.model.Ambiti;

import java.sql.Date;
import java.util.List;

public interface Annuncio {

    public Annuncio findById(Long id);

    public List<Annuncio> findByTitolo(String titolo);

    public List<Annuncio> findByDescrizione(String descrizione);

    public List<Annuncio> findByDataDiScadenza(Date dataDiScadenza);

    public List<Annuncio> findByProvinciaAnnuncio(String provinciaAnnuncio);
    public List<Annuncio> findByAmbiti(List<Ambiti> ambiti);

    public List<Annuncio> findAll();

    public void save(Annuncio annuncio);

    public void update(Annuncio annuncio);

    public void delete(Annuncio annuncio);
}

File: SkillMatch/src/app/model/Annuncio.ts
import { Ambito } from "./Ambito";
import { Cliente } from "./Cliente";
import { Proposta } from "./Proposta";
import {Province} from "./Province";

export interface Annuncio{

    id: number ;
    titolo: string;
    descrizione: string;
    dataDiScadenza: Date;
    provinciaAnnuncio: Province | string;
    image: string;
    cliente: Cliente;
    ambito: Ambito;
    proposta: Proposta;
    consigliato: boolean
    starClicked: boolean

}
